app/src/main/java/com/marvelapp/presentation/activity/CharacterActivity.kt

```diff
@@ -0,0 +1,50 @@
+package com.marvelapp.presentation.activity
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.marvelapp.data.CharacterEntity
+import com.marvelapp.databinding.ActivityCharacterBinding
+import com.marvelapp.presentation.adapter.CharacterAdapter
+import com.marvelapp.presentation.viewmodel.CharacterViewModel
+import com.marvelapp.presentation.viewmodel.CharacterViewModel.CharactersState.SHOW_CHARACTERS
+import com.marvelapp.presentation.viewmodel.CharacterViewModel.CharactersState.WAITING_CHARACTERS
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collectLatest
+
+@AndroidEntryPoint
+class CharacterActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityCharacterBinding
+    private val viewModel: CharacterViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityCharacterBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        lifecycleScope.launchWhenStarted {
+            viewModel.state.collectLatest { this@CharacterActivity.updateUI(it) }
+        }
+        viewModel.showCharacters()
+    }
+
+    private fun updateUI(state: CharacterViewModel.CharactersData) {
+        when (state.state) {
+            WAITING_CHARACTERS -> {} // DO NOTHING
```

Comment:
why do you have this DO NOTHING?

Reply:
as it is for the default value needed by the stateFlow, you can rename the state to INIT and call `viewModel.showCharacters()` here

Reply:
Solved